### PR TITLE
add command to show pkgs with updates available

### PR DESCRIPTION
**Pull request by maks**
This PR adds a command that will list for all the dependencies in the local pubspec.yaml the current packages and the latest available on pub.dev to quickly give a summary of which packages have updated versions available:

```
dart bin/pubx.dart u
http    [^0.12.0]       latest: 0.12.0+2
path    [^1.6.0]        latest: 1.6.4
yaml    [^2.1.16]       latest: 2.1.16
quiver  [^2.0.4]        latest: 2.0.5
args    [^1.5.0]        latest: 1.5.2
```

### CODE REVIEW

File: lib/src/commands/updates.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    // Load and parse pubspec.yaml
+    final pubContents = await pub.readAsStringSync();
+    final pubYaml = loadYaml(
+      pubContents,
+      sourceUrl: pub.uri,
+    ) as YamlMap;
+
+    final YamlMap dependencies = pubYaml['dependencies'];
+
+    dependencies.keys.forEach((packageName) async {
```
Comment (shyndman) [inline lib/src/commands/updates.dart]: nit: Mind making this a for...in?

File: lib/src/commands/updates.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      if (!dependencies[packageName].toString().startsWith('{sdk:')) {
+        try {
+          final pkgInfo = await fetchPackageInfo(packageName);
+          print(
+              '${pkgInfo.name} [${dependencies[packageName]}]  latest: ${pkgInfo.version}');
```
Comment (shyndman) [inline lib/src/commands/updates.dart]: I notice in the PR description the strings are aligned in columns, but don't appear to be here.

File: README.md [hunk 3]
```diff
@@ -36,6 +36,14 @@ Searches [pub.dev](https://pub.dev).
 
 ### view
```
Comment (shyndman) [inline README.md]: This should be changed to "updates"

File: README.md [hunk 4]
```diff
@@ -36,6 +36,14 @@ Searches [pub.dev](https://pub.dev).
...
 
+*Aliases: updates, u*
```
Comment (shyndman) [inline README.md]: nit: Technically "updates" isn't an alias here.

**Review by shyndman — CHANGES_REQUESTED** [review]
Thank you for the contribution. This looks great. :)

Just a few nits.

File: lib/src/commands/updates.dart (hunk 2 above)
Comment (shyndman) [reply lib/src/commands/updates.dart]: It's fine if this is intentional. We can always evolve it. Just curious whether this was intentional.

File: README.md (hunk 4 above)
Comment (maks) [reply README.md]: oops, sorry thats a copy-paste typo, will fix that.

File: lib/src/commands/updates.dart (hunk 2 above)
Comment (maks) [reply lib/src/commands/updates.dart]: yes sorry about that, I intended for them aligned but didn't get time to do that when I first wrote this PR, I'll have a go at getting that done now.

Comment (maks) [reply lib/src/commands/updates.dart]: I've reworked this a bit so the output should be correctly aligned, taking into account package name lengths and hosted vs non-hosted packages, eg.

```
provider               	 [^3.0.0+1] 	 latest: 4.0.2
uuid                   	 [^2.0.1] 	 latest: 2.0.4
device_info            	 [^0.4.0] 	 latest: 0.4.1+4
package_info           	 [^0.4.0] 	 latest: 0.4.0+13
http                   	 [path pkg] 	 latest: 0.12.0+4
sentry                 	 [git pkg] 	 latest: 3.0.1
flutter_markdown       	 [git pkg] 	 latest: 0.3.3
url_launcher           	 [^5.1.4] 	 latest: 5.4.1
google_sign_in         	 [^4.0.1+3] 	 latest: 4.1.1
retry                  	 [^3.0.0] 	 latest: 3.0.0+1
cupertino_icons        	 [^0.1.2] 	 latest: 0.1.3
flutter_facebook_login 	 [^3.0.0] 	 latest: 3.0.0

```

**Comment by maks** [comment]
@shyndman thanks for taking a look at the PR and all the great feedback!
I've hopefully addressed all the issues in the latest set of commits

File: lib/src/commands/updates.dart [hunk 5]
```diff
@@ -30,16 +30,43 @@ class UpdatesCommand extends Command {
...
           //NA
         }
       }
-    });
+    }
+  }
+
+  bool _isSdkPackage(YamlMap package) => package['sdk'] != null;
+
+  String _packageType(YamlMap package) {
```
Comment (shyndman) [inline lib/src/commands/updates.dart:63]: Ah, edge cases. Great. :)

File: lib/src/commands/updates.dart (hunk 2 above)
Comment (shyndman) [reply lib/src/commands/updates.dart]: 👍

**Comment by shyndman** [comment]
Merged. Thanks for the contribution. I'll put up a new version shortly.

**Pull request merged**